coordinator/internal/meshapi/meshapi.go

```diff
@@ -100,3 +102,67 @@ func (i *Server) NewMeshCert(ctx context.Context, _ *meshapi.NewMeshCertRequest)
 
 	return resp, nil
 }
+
+// Recover verifies the peers policy and role and recovers the calling Coordinator.
+func (i *Server) Recover(ctx context.Context, _ *meshapi.RecoverRequest) (*meshapi.RecoverResponse, error) {
+	i.logger.Info("Recover called")
+
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, fmt.Errorf("failed to get peer from context")
+	}
+
+	authInfo, ok := p.AuthInfo.(authority.AuthInfo)
+	if !ok {
+		return nil, fmt.Errorf("unexpected AuthInfo type: %T", p.AuthInfo)
+	}
+	state := authInfo.State
+	report := authInfo.Report
+
+	hostData := manifest.NewHexString(report.HostData())
+	entry, ok := state.Manifest().Policies[hostData]
+	if !ok {
+		return nil, status.Errorf(codes.PermissionDenied, "policy hash %s not found in manifest", hostData)
+	}
+	if entry.Role != manifest.RoleCoordinator {
+		return nil, status.Errorf(codes.PermissionDenied, "role %s not allowed to recover", entry.Role)
```

Comment:
```suggestion
		return nil, status.Errorf(codes.PermissionDenied, "role %q not allowed to recover", entry.Role)
```
Role may be empty, so quoting it may help readability.